security/pkg/nodeagent/tools/sds_cli.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"time"
+
+	xdsapi "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	disapi "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v2"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/security/pkg/nodeagent/sds"
+)
+
+var fakeCredentialToken = "fakeToken"
+
+func sdsRequest(socket string, req *xdsapi.DiscoveryRequest) *xdsapi.DiscoveryResponse {
+	var opts []grpc.DialOption
+
+	opts = append(opts, grpc.WithInsecure())
+	opts = append(opts, grpc.WithDialer(func(addr string, timeout time.Duration) (net.Conn, error) {
+		return net.DialTimeout("unix", socket, timeout)
+	}))
+
+	conn, err := grpc.Dial(socket, opts...)
+	if err != nil {
+		panic(err.Error())
+	}
+	defer conn.Close()
+
+	sdsClient := disapi.NewSecretDiscoveryServiceClient(conn)
+	header := metadata.Pairs(sds.CredentialTokenHeaderKey, fakeCredentialToken)
+	ctx := metadata.NewOutgoingContext(context.Background(), header)
+	stream, err := sdsClient.StreamSecrets(ctx)
+	if err != nil {
+		panic(err.Error())
+	}
+	err = stream.Send(req)
+	if err != nil {
+		panic(err.Error())
+	}
+	res, err := stream.Recv()
+	if err != nil {
+		panic(err.Error())
+	}
+	return res
+}
+
+func main() {
+	sdsSocket := flag.String("socket", "/tmp/sdsuds.sock", "SDS socket")
```

Comment:
Should the default value matches against the node_agent_k8s/main.go?

Reply:
yes they should match.